src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -398,13 +415,110 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
+        status=None,
+        platform=PlatformUpdateParameters(
+            os=None
+        ),
+        agent_configuration=AgentProperties(
+            cpu=None
+        ),
+        timeout=None,
+        step=None,
+        trigger=TriggerUpdateParameters(
+            source_triggers=None,
+            base_image_trigger=None
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_identity_remove(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             identities=None,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if identities is not None and IDENTITY_GLOBAL_REMOVE in identities:
+        if len(identities) > 1:
+            raise CLIError(
+                "Cannot specify additional identities when [all] is used in [--identities]")
+
+    identity = None
+    if not identities or IDENTITY_LOCAL_ID in identities:
+        # To remove only the system assigned identity if user-assigned identities also exist
+        # PATCH with the existing user-assigned identities
+        # If no user-assigned identities exist, set the type to None
+        existingUserIdentites = client.get_details(
+            resource_group_name, registry_name, task_name).identity.user_assigned_identities
+        identities = IDENTITY_GLOBAL_REMOVE if not existingUserIdentites else list(existingUserIdentites.keys())
+        identity = _build_identities_info(cmd, identities)
+    else:
+        identity = _build_identities_info(cmd, identities, True)
+
+    TaskUpdateParameters, PlatformUpdateParameters, AgentProperties, TriggerUpdateParameters = cmd.get_models(
+        'TaskUpdateParameters', 'PlatformUpdateParameters', 'AgentProperties', 'TriggerUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        identity=identity,
+        status=None,
+        platform=PlatformUpdateParameters(
+            os=None
+        ),
+        agent_configuration=AgentProperties(
+            cpu=None
+        ),
+        timeout=None,
+        step=None,
+        trigger=TriggerUpdateParameters(
+            source_triggers=None,
+            base_image_trigger=None
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_identity_show(cmd,
+                           client,
+                           task_name,
+                           registry_name,
+                           resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    resp = client.get_details(resource_group_name, registry_name, task_name).identity
+
+    return {} if not resp else resp
```

Comment:
Curious why need to explicitly return `{}`. Can we just return resp like other commands?

Reply:
Its what we do in CLI. If the RP returns Null. The experience will be that when they do az acr show something, nothing gets printed. 
v/s if you do {} they know they dont have anything to show. We do this in couple other places.

Reply:
@northtyphoon I was keeping consistent with the existing credential command group return. However, I can change this back. Do we want to also change the credential return? //cc @shahzzzam

Reply:
I remember we had a conversation about this and we settled on using `{}` or `[]` to keep consistent with other ACR commands. //cc @djyou

Reply:
make sense

Reply:
I would prefer returning server response as is. It's consistent to show nothing if the result is empty. If `{}` is preferred, I would question why server side is not returning empty dict but null.

Reply:
There are multiple cases where Identity is null server side. For instance, if a call is made from an older API version where Identity is not defined. We sanitize the Task based on API version and this property will always be null for any lower api version than 2018-09-01 when GET is called. //cc @mnltejaswini

Reply:
The CLI will only make request to the latest api-version 2018-09-01?